featuretools/feature_base/feature_base.py

```diff
@@ -602,12 +612,14 @@ def __init__(self, base_features, primitive):
                                                base_features=base_features,
                                                relationship_path=[],
                                                primitive=primitive)
+        if name is not None:
+            self._name = name
 
     @classmethod
     def from_dictionary(cls, arguments, entityset, dependencies, primitives_deserializer):
         base_features = [dependencies[name] for name in arguments['base_features']]
         primitive = primitives_deserializer.deserialize_primitive(arguments['primitive'])
-        return cls(base_features, primitive)
+        return cls(base_features, primitive, arguments['name'])
```

Comment:
`return cls(base_features=base_features, primitive=primitive, name=arguments['name'])` so we specify what the arguments are.